proto/internal/temporal/server/api/deployment/v1/message.proto

```diff
@@ -31,47 +31,72 @@ import "google/protobuf/timestamp.proto";
 import "temporal/api/deployment/v1/message.proto";
 import "temporal/api/common/v1/message.proto";
 
+// Data for each deployment+task queue pair. This is stored in each deployment (for each task
+// queue), and synced to task queue user data (for each deployment).
+message Data {
+    google.protobuf.Timestamp first_poller_time = 1;
+    google.protobuf.Timestamp last_became_current_time = 2;
+}
+
 message DeploymentLocalState {
     temporal.api.deployment.v1.Deployment worker_deployment = 1;
     map<string, TaskQueueFamilyInfo> task_queue_families = 2;
     bool is_current = 3;
     google.protobuf.Timestamp create_time = 4;
-    map<string, temporal.api.common.v1.Payload> metadata = 5; 
-    
+    map<string, temporal.api.common.v1.Payload> metadata = 5;
+
     message TaskQueueFamilyInfo {
         // key: taskQueueType, val: TaskQueueInfo
-        map<int32,TaskQueueInfo> task_queues = 1;
-        
+        map<int32, TaskQueueInfo> task_queues = 1;
+
         message TaskQueueInfo {
             temporal.api.enums.v1.TaskQueueType task_queue_type = 1;
-            google.protobuf.Timestamp first_poller_time = 2;
+            Data data = 2;
```

Comment:
Conceptually, `data.last_became_current_time` is filled based on deployment's `last_became_current_time`. Once we have Rollout, deployment's `last_became_current_time` will come from rollout records. For now, we should put the `last_became_current_time` inside `DeploymentLocalState` and update is together with `is_current`.

Given that, maybe `Data` should not be stored in here and only be calculated from the DeploymentLocalState + TaskQueueInfo when we call `SyncUserData`. @dnr wdyt?

Reply:
So add the last_became_current_time at top level but still use `Data` inside the state per TQ? I guess I don't have an objection, we can see if they'll converge more in the future and if so we can split the proto defs.

Reply:
yes, that's what I was thinking. I could be wrong, it depends how it evolves.